### ISSUE TITLE
custom theme extended from alabaster or sphinx_rtd_theme can't find base theme.

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1794]
Sphinx 1.3 include theme paths for 'alabaster' and 'sphinx_rtd_theme' automatically if `html_theme` is specified them.
However, if user custom theme extends from them, the path isn't included automatically.

IMO, a custom theme has `inherit = alabaster` or `inherit = sphinx_rtd_theme`, the base theme path should be included automatically.

**Comment by shimizukawa** [comment]
Now sphinx theme loader loads alabaster and sphinx_rtd_theme automatically if a theme is inherited from them.This ticket can be closed.

However, alabaster and sphinx_rtd_theme doesn't have theme name directory.
Consequence of it, layout.html of custom derived theme can't extend from them because `{%- extends "alabaster/layout.html" %}` is not valid and `{%- extends "!layout.html" %}` cause infinite loop (leadning exclamation mark excludes `_templates`, but doesn't exclude custom theme directory itself).

In conclusion, themes without a theme name directory in the theme path (like alabaster-0.7.3 and sphinx_rtd_theme-0.1.7) is not extendable.

**Comment by shimizukawa** [comment]
> However, alabaster and sphinx_rtd_theme doesn't have theme name directory.
> Consequence of it, layout.html of custom derived theme can't extend from them because `{%- extends "alabaster/layout.html" %}` is not valid and `{%- extends "!layout.html" %}` cause infinite loop (leadning exclamation mark excludes `_templates`, but doesn't exclude custom theme directory itself).
> 
> In conclusion, themes without a theme name directory in the theme path (like alabaster-0.7.3 and sphinx_rtd_theme-0.1.7) is not extendable.

I was wrong.
Now sphinx supports theme that inherit from non-builtin themes.
